yt/frontends/gamer/fields.py

```diff
@@ -59,42 +69,110 @@ def setup_fluid_fields(self):
         unit_system = self.ds.unit_system
         unit_system.registry = self.ds.unit_registry  # TODO: Why do I need this?!
 
+        temp_conv = pc.kb / (self.ds.mu * pc.mh)
+
         if self.ds.srhd:
             c2 = pc.clight * pc.clight
+            invc2 = 1.0 / c2
```

Comment:
Similarly, how about setting all physical constants to unity for dimensionless simulations? For example,
```
            if self.ds.unit:
                c2 = pc.clight * pc.clight
                invc2 = 1.0 / c2
                c = pc.clight.in_units("code_length / code_time")
            else:
                c = self.ds.arr(1.0, "code_velocity")
                c2 = c*c
                invc2 = 1.0 / c2
```
Together with `temp_conv`, they ensure no extra unit conversion in dimensionless simulations. I've confirmed that these modifications are necessary to get the correct results when computing, for example, the effective adiabatic index `("gas", "gamma")` in the `Riemann` test problem.

Reply:
Done.